query-service-impl/src/main/java/org/hypertrace/core/query/service/projection/ProjectionTransformation.java

```diff
@@ -358,43 +358,26 @@ private QueryRequest rebuildRequest(
   private Filter rebuildFilterForComplexAttributeExpression(
       Filter originalFilter, List<OrderByExpression> orderBys) {
 
-    Filter filterFromOrderBys = createFilterForComplexAttributeExpressionFromOrderBy(orderBys);
     Filter updatedFilter = updateFilterForComplexAttributeExpressionFromFilter(originalFilter);
+    List<Filter> childFilterList = createFilterForComplexAttributeExpressionFromOrderBy(orderBys);
 
-    if (!updatedFilter.equals(Filter.getDefaultInstance())
-        && !filterFromOrderBys.equals(Filter.getDefaultInstance())) {
-      return Filter.newBuilder()
-          .setOperator(Operator.AND)
-          .addChildFilter(updatedFilter)
-          .addChildFilter(filterFromOrderBys)
-          .build();
-    } else if (!filterFromOrderBys.equals(Filter.getDefaultInstance())) {
-      if (filterFromOrderBys.getChildFilterCount() == 1) {
-        return filterFromOrderBys.getChildFilter(0);
+    if (childFilterList.isEmpty()) {
+      return updatedFilter;
+    } else {
+      if (!updatedFilter.equals(Filter.getDefaultInstance())) {
+        childFilterList.add(updatedFilter);
```

Comment:
Assuming a list another function returns is mutable is dangerous - it's correct right now, but it's an easy bug if someone later tightens up the downstream code to use an immutable list instead.

Reply:
resolved